### ISSUE TITLE
Compilation errors in Swift 3.0

### ISSUE THREAD

**Issue by igormatyushkin014** [MatthewYork/DateTools#173]
I'm using `swift` branch via CocoaPods:

```
pod 'DateTools', :git => 'https://github.com/MatthewYork/DateTools', :branch => 'swift'
```

When I compile project, I'm getting several compilation errors:
- Value of type `Date` has no member `subtract`
- Value of type `Date` has no member `add`
- Cannot convert value of type `TimeChunk` to expected argument type `TimeInterval` (aka `Double`)

![2016-10-02 0 21 45](https://cloud.githubusercontent.com/assets/8403399/19017079/f67bbca2-8836-11e6-8202-343a8f89ae0f.png)

**Comment by k3zi** [comment]
+1, I am having the same issue, but a week ago one of my other projects compiled with this branch. Possibly a bug was introduced since then?

**Comment by MatthewYork** [comment]
That is entirely possible. We are tying to keep the branch as stable as we can, but there is a lot of active development on it as it is prerelease.

To see what is working correctly, I recommend running the unit projects on the swift branch. We will do our best resolve the cocoa pods issue and to not break the branch in the future.

**Comment by dianambb** [comment]
I saw this too. Use tag in the cocoapod instead of the branch.

`pod 'DateTools', :git => 'https://github.com/MatthewYork/DateTools.git', :tag => '2.0.0-beta.1'`

**Comment by MatthewYork** [comment]
This is the correct answer. Also, as of pretty much now, beta 2 is available on the swift branch.

The tag is: `2.0.0-beta.2`

I will update the pod spec to account for this change.

**Comment by MatthewYork** [comment]
@k3zi @igormatyushkin014 Please let me know if this resolved your issue. Thanks!

**Comment by pushian** [comment]
<img width="320" alt="screen shot 2016-10-05 at 8 13 01 pm" src="https://cloud.githubusercontent.com/assets/16488850/19112576/2c4fb9d8-8b38-11e6-81cb-2667fa6338a2.png">
I am using 2.0.0-beta.2, but i will still get the above error.

**Comment by codemodouk** [comment]
@pushian - Has "isEarlierThan" been replaced by "less" method for the Swift branch?

**Comment by pushian** [comment]
@codemodouk I have found the answer. "isEarlierThan" has been replaced by "earlier(than: ***)"

**Comment by pushian** [comment]
"addingDays" function is unavailable now. The most similar function I can find is add(timeChunk: TimeChunk) but I cannot figure out how to use this function. Anyone please help.

**Comment by MatthewYork** [comment]
@codemodouk @pushian So, we have a lot of new functionality that is yet undocumented on the Swift branch... The `add()` method will consume a time chunk, which is an all new high-level extension of DateComponents that allows for grouping and shorthanding complex component operations. For instance, A time chunk can be constructed as `2.days` or `4.weeks` or `3.minutes + 2.seconds`. This may then be applied as date component updates to a given date, time period, chain, etc. Time intervals are also supported to give you both fuzzy and non-fuzzy option when manipulating dates.

There will be a *lot* of new documentation once we make another few passes through the API to ensure consistency. This iteration had multiple writers, so I am sure there is much to unify. Until then, I will try to answer any questions that arise!

**Comment by MatthewYork** [comment]
Also, `earlier(than:)` should be `isEarler(than:)`. You will see all boolean returning functions prefixed by "is" and "has". Part of that unification I was mentioning.

**Comment by MatthewYork** [comment]
These issues will be marked as resolved for version 2.0.2. Please reopen if there are any new developments. Regards, Matt.
